tests/auto/tst_declarativebookmarkmodel/tst_declarativebookmarkmodel.cpp

```diff
@@ -9,211 +9,227 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include <QObject>
 #include <QtTest>
-#include <QtQml>
+#include <QStandardPaths>
+#include <QDir>
 #include <QFile>
-#include <QDesktopServices>
-#include <QGuiApplication>
+#include <QTextStream>
+
 #include "declarativebookmarkmodel.h"
-#include "testobject.h"
 #include "bookmarkmanager.h"
 
-static const QByteArray QML_SNIPPET = \
-        "import QtQuick 2.0\n" \
-        "import Sailfish.Browser 1.0\n" \
-        "Item {\n" \
-        "   width: 100; height: 100\n" \
-        "   property alias bookmarkModel: model\n" \
-        "   BookmarkModel { id: model }\n" \
-        "}\n";
+static const QByteArray BOOKMARKS_JSON = \
+    "[{" \
+    "  \"favicon\": \"image://theme/icon-launcher-jollacom\"," \
+    "  \"hasTouchIcon\": true," \
+    "  \"title\": \"Jolla\"," \
+    "  \"url\": \"http://jolla.com/\"" \
+    "}]";
 
+static const QString JOLLA_URL = "http://jolla.com/";
+static const QString TEST_URL = "http://www.test1.jolla.com";
 
-class tst_declarativebookmarkmodel : public TestObject
+class tst_declarativebookmarkmodel : public QObject
 {
     Q_OBJECT
 
-public:
-    tst_declarativebookmarkmodel();
-
 private slots:
     void initTestCase();
-    void cleanupTestCase();
+    void init();
+    void cleanup();
 
-    void addBookmark();
-    void removeBookmark();
-    void contains();
-    void editBookmark();
-    void activeUrlBookmarked();
+    void add();
+    void remove();
     void removeByIndex();
+    void contains();
+    void edit();
+    void setActiveUrl();
     void clearBookmarks();
+    void updateFavoriteIcon();
+    void data();
 
 private:
-    DeclarativeBookmarkModel *bookmarkModel;
+    QPointer<DeclarativeBookmarkModel> m_model;
+    QString m_bookmarksFile;
 };
 
-
-tst_declarativebookmarkmodel::tst_declarativebookmarkmodel()
-    : TestObject(QML_SNIPPET)
+void tst_declarativebookmarkmodel::initTestCase()
 {
-    bookmarkModel = TestObject::qmlObject<DeclarativeBookmarkModel>("bookmarkModel");
+    QString settingsLocation = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
```

Comment:
Maybe worth adding in another PR something like ```BrowserPaths``` class that would have static methods like.

```
static QString downloadLocation()
static QString picturesLocation()
static QString dataLocation()
...
```
This way we could clean up a bit the webutils thingie and unit test cases would not contain assumptions about paths.